### PR TITLE
Add negative and edge case tests for MakeMove and UnmakeMove

### PR DESCRIPTION
**Pull request by RchrdHndrcks**
## Summary
- **TestMakeMove_InvalidMoves**: Table-driven tests verifying that illegal moves (same-square, out-of-range, opponent's piece, empty square, gibberish strings) are all rejected with an error.
- **TestUnmakeMove_NoHistory**: Verifies that calling UnmakeMove on a fresh game does not panic and leaves the game state completely unchanged (FEN, turn, available moves, check/checkmate/stalemate flags).
- **TestAvailableMoves_Consistency**: After each move in a short opening sequence, confirms that every move returned by AvailableMoves() can actually be played via MakeMove(), and that moves not in AvailableMoves() are rejected.

## Test plan
- [x] All new tests pass with `go test ./...`
- [x] Existing tests remain green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: chess/chess_test.go [hunk 1]
```diff
@@ -910,3 +911,95 @@ func TestUnmakeMove(t *testing.T) {
...
+}
+
+func TestUnmakeMove_NoHistory(t *testing.T) {
+	// Arrange
+	c, err := chess.New()
+	require.NoError(t, err)
+
+	fenBefore := c.FEN()
+	turnBefore := c.Turn()
+	movesBefore := c.AvailableMoves()
```
[the remark this answers is not in the document]
Comment (RchrdHndrcks) [reply chess/chess_test.go:1020]: This concern is not applicable here. The test `TestUnmakeMove_NoHistory` specifically verifies that `UnmakeMove()` on a fresh game is a no-op -- no state mutation occurs at all. Since nothing changes, it does not matter whether `movesBefore` holds the same reference as the internal slice; they will trivially be equal. If a future implementation were to mutate the internal slice during a no-op `UnmakeMove()`, the test would already be broken in a more fundamental way, and cloning would just mask the real problem.

File: chess/chess_test.go [hunk 2]
```diff
@@ -910,3 +911,95 @@ func TestUnmakeMove(t *testing.T) {
...
+		available := c.AvailableMoves()
+		require.NotNil(t, available, "AvailableMoves should never return nil")
+
+		// Verify every available move can actually be played (use a clone-like approach).
+		for _, m := range available {
+			clone, err := chess.New(chess.WithFEN(c.FEN()))
+			require.NoError(t, err, "failed to create game from FEN %s", c.FEN())
+
+			err = clone.MakeMove(m)
+			assert.NoError(t, err, "available move %q should be playable after move %q", m, move)
```
[the remark this answers is not in the document]
Comment (RchrdHndrcks) [reply chess/chess_test.go:1059]: This is premature optimization. The test iterates over only 5 positions, each with roughly 20-30 legal moves. The total number of clones created is small, and the test runs in well under a second even with the default parallelism. Adding `WithParallelism(1)` would add noise to the test code for no practical benefit. If performance ever became a concern for this test (unlikely), it could be addressed then.

**Comment by RchrdHndrcks** [comment]
Merge conflicts
